packages/shared/src/graphql/integrations.ts

```diff
@@ -0,0 +1,127 @@
+import { gql } from 'graphql-request';
+import type { Source } from './sources';
+
+export enum UserIntegrationType {
+  Slack = 'slack',
+}
+
+export type UserIntegration = {
+  id: string;
+  type: UserIntegrationType;
+  createdAt: Date;
+  updatedAt: Date;
+  name: string;
+  userId: string;
+};
+
+export type SlackChannel = {
+  id: string;
+  name: string;
+};
+
+export type UserSourceIntegration = {
+  type: UserIntegrationType;
+  createdAt: Date;
+  updatedAt: Date;
+  userIntegration: UserIntegration;
+  source: Source;
+  channelIds: string[];
+};
+
+export type SlackChannelConnection = {
+  data: SlackChannel[];
+};
+
+export const SLACK_CHANNELS_QUERY = gql`
+  query SlackChannels($integrationId: ID!) {
+    slackChannels(integrationId: $integrationId, limit: 999) {
```

Comment:
Is the `limit: 999` intended to be that high?

Reply:
and for https://github.com/dailydotdev/apps/pull/3315#issuecomment-2277915616 I did not really touch dropdown component so it looks as it looks I guess 🙈

Reply:
I'll take a quick look if I can fix it though

Reply:
basically when `scrollable` is set on dropdown it gets fixed height, recon is something to do with styling so leaving it as is.